lib/internal/modules/esm/assets/load.js

```diff
@@ -0,0 +1,300 @@
+        }
+      }
+      if (res.statusCode > 303 || res.statusCode < 200) {
+        dispose();
+        reject(
+          new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+            parsed.href,
+            'HTTP response returned status code of ' + res.statusCode));
+        return;
+      }
+      const { headers } = res;
+      const contentType = headers['content-type'];
+      if (!contentType) {
+        dispose();
+        reject(new ERR_UNSUPPORTED_ESM_NETWORK_BAD_RESPONSE(
+          parsed.href,
+          'content-type is required'));
+        return;
+      }
+      /**
+       * @type {CacheEntry}
+       */
+      const entry = {
+        resolvedHREF: parsed.href,
+        headers: {
+          'content-type': res.headers['content-type']
+        },
+        body: new Promise((f, r) => {
+          const buffers = [];
+          let size = 0;
+          let bodyStream = res;
+          let onError;
+          if (res.headers['content-encoding'] === 'br') {
+            bodyStream = createBrotliDecompress();
+            onError = function onError(e) {
+              bodyStream.close();
+              dispose();
+              reject(e);
+              r(e);
+            };
+            res.on('error', onError);
+            res.pipe(bodyStream);
+          } else if (res.headers['content-encoding'] === 'gzip' ||
+            res.headers['content-encoding'] === 'deflate') {
+            bodyStream = createUnzip();
+            onError = function onError(e) {
+              bodyStream.close();
+              dispose();
+              reject(e);
+              r(e);
+            };
+            res.on('error', onError);
+            res.pipe(bodyStream);
+          } else {
+            onError = function onError(e) {
+              dispose();
+              reject(e);
+              r(e);
+            };
+          }
+          bodyStream.on('error', onError);
+          bodyStream.on('data', (d) => {
+            ArrayPrototypePush(buffers, d);
+            size += d.length;
+          });
+          bodyStream.on('end', () => {
+            const body = entry.body = /** @type {Buffer} */(
+              BufferConcat(buffers, size)
+            );
+            f(body);
+          });
+        }),
+      };
+      getCache.set(parsed.href, entry);
+      fulfill(entry);
+    });
+  });
+  getCache.set(parsed.href, result);
+  return result;
+}
+
+/**
+ * Returns if an address has local status by if it is going to a local
+ * interface or is an address resolved by DNS to be a local interface
+ * @param {string} hostname url.hostname to test
+ * @returns {Promise<boolean>}
+ */
+async function isLocalAddress(hostname) {
+  try {
+    if (StringPrototypeStartsWith(hostname, '[') &&
+        StringPrototypeEndsWith(hostname, ']')) {
+      hostname = StringPrototypeSlice(hostname, 1, -1);
+    }
+    const addr = await dnsLookup(hostname, { verbatim: true });
+    if (addr.family === 4) {
+      return addr.address === '127.0.0.1';
```

Comment:
Per [RFC5735](https://datatracker.ietf.org/doc/html/rfc5735), entire `127.0.0.0/8` is used for loopback. So why restrict it to be `127.0.0.1`? Just curious.

Reply:
No real reason, could match the whole range without issues to my knowledge

Reply:
Sorry, just to be clear, `127.0.0.0/8` means `127.0.0.[0–8]` (`127.0.0.0`, `127.0.0.1`, …, `127.0.0.8`)?

Reply:
> Sorry, just to be clear, `127.0.0.0/8` means `127.0.0.[0–8]` (`127.0.0.0`, `127.0.0.1`, …, `127.0.0.8`)?

Although the question seems to be off-topic, I will answer it anyway. `127.0.0.0/8` means from `127.0.0.0` to `127.255.255.255`. And it actually makes the match a lot easier:

```diff
    if (addr.family === 4) {
-      return addr.address === '127.0.0.1';
+     return StringPrototypeStartsWith(addr.address, '127.');
```